kitsune/users/tests/test_api.py

```diff
@@ -187,6 +187,12 @@ class TestUserView(TestCase):
     def setUp(self):
         self.client = APIClient()
 
+    def test_usernames_with_periods(self):
+        p = ProfileFactory(user__username='something.something')
+        url = reverse('user-detail', args=[p.user.username])
+        res = self.client.get(url)
+        eq_(res.status_code, 200)
```

Comment:
You might want to also verify that the username is in the data that comes back. Given that this is an API, that's possibly JSON. You might need to also send an Accept header. Verifying that the data that comes back contains the username guarantees you that this actually worked rather than returning something else.